### ISSUE TITLE
Problem on a number column

### ISSUE THREAD

**Issue by TanguyPa** [pnp/pnpcore#751]
## Category

- [x] Bug

## Describe the bug

I try to read the data of an item that has a column of type "Number", when the latter is filled the right value is returned.

Except that when it is not filled, the value 0 is returned, so I can't distinguish an item with a value of 0 from an item that has no value.

I'm using LoadListDataAsStreamAsync with camlquery, the item has the content type "Document set".

## Steps to reproduce
I get the items this way (filter to differentiate between doc sets and files)

```csharp
string viewXml = @"<View Scope='RecursiveAll'>
										<ViewFields>
										  <FieldRef Name='MyNumberColumn' />          
                                                                                  <FieldRef Name='FileLeafRef' />                                           
                                                                                  <FieldRef Name='FileDirRef' />                                           
                                                                                  <FieldRef Name='FileRef' />            
                                                                                  <FieldRef Name='Title' />                                    
										</ViewFields>                                              
									   </View>";
// Execute the query
var output = await list.LoadListDataAsStreamAsync(new PnP.Core.Model.SharePoint.RenderListDataOptions()
{
	ViewXml = viewXml,
	RenderOptions = RenderListDataOptionsFlags.ListData
});
foreach (var item in list.Items.AsRequested())
{
	if ((string)item["ContentTypeId"] == "xxxx")
	{
		ensemblesDocsItems.Add(item);
	}
	else if ((string)item["ContentTypeId"] == "xxxx")
	{
		docsItems.Add(item);
	}

}
```
```csharp
foreach (var ensItem in ensemblesDocsItems)
{                       
     var value = ensItem["MyNumberColumn"];
}
```

## Expected behavior

ensItem["MyNumberColumn"] == 10 when the value is 10
ensItem["MyNumberColumn"] == 0 when the value is 0
ensItem["MyNumberColumn"] == null (or something else) when there is no value

## Environment details (*development & target environment*)
SDK version: 1.5.0 (try 1.4.0 too)
OS: Windows 10
SDK used in: Console App
Framework: .NET 5.0
Tooling: Visual Studio 2019

**Comment by jansenbe** [comment]
@TanguyPa , good catch. When using `LoadListDataAsStreamAsync` this indeed goes wrong. Fix will be checked in soon

**Comment by jansenbe** [comment]
@TanguyPa : and the fix is in. Please try again with our next nightly build that will be available as of tomorrow (you need version 1.5.51 or higher) and let us know if things work as expected. Thanks for reporting this issue!

**Comment by TanguyPa** [comment]
Thank you, it works fine
